src/QuadForm/Herm/Genus.jl

```diff
@@ -1322,6 +1345,8 @@ function direct_sum(G1::HermGenus, G2::HermGenus)
   sig1 = G1.signatures
   sig2 = G2.signatures
   g3 = merge(+, sig1, sig2)
+  bd = union(support(2*maximal_order(base_field(E))), support(discriminant(maximal_order(E))))
+  filter!(g -> (prime(g) in bd) || (scales(g) != Int[0]), LGS)
```

Comment:
Is the `scales(g) != Int[0]` for the block not being unimodular?

Reply:
Exactly, we keep the local symbol if the prime of bad, or if it is not unimodular. A local symbol is unimodular if and only if it consists of only one block of scale zero.